src/machine/machine_rp2040_usb.go

```diff
@@ -169,21 +169,19 @@ func initEndpoint(ep, config uint32) {
 }
 
 func handleUSBSetAddress(setup usb.Setup) bool {
+	rp.USBCTRL_REGS.SIE_STATUS.Set(rp.USBCTRL_REGS_SIE_STATUS_ACK_REC)
 	sendUSBPacket(0, []byte{}, 0)
-
 	// last, set the device address to that requested by host
 	// wait for transfer to complete
-	timeout := 3000
-	rp.USBCTRL_REGS.SIE_STATUS.Set(rp.USBCTRL_REGS_SIE_STATUS_ACK_REC)
+	start := rp.TIMER.TIMERAWL.Get()
 	for (rp.USBCTRL_REGS.SIE_STATUS.Get() & rp.USBCTRL_REGS_SIE_STATUS_ACK_REC) == 0 {
-		timeout--
-		if timeout == 0 {
-			return true
+		current := rp.TIMER.TIMERAWL.Get()
+		elapsed := current - start //Avoiding overflow in unsigned integer subtraction.
```

Comment:
This is subtle, so I suggest moving it to a (unexported) method on `timerType`, say

```go
// delay for the specified time in milliseconds,
// up to xx minutes.             <---- 72 minutes?
func (tmr *timerType) delay(us uint32)
```

Reply:
Alternatively, if you don't mind the slower uint64 implementation, use the already portable `timer.timeElapsed()` instead of `rp.TIMER.TIMERAWL`.